zio-http/src/main/scala/zhttp/service/DecodeJResponse.scala

```diff
@@ -11,6 +12,8 @@ trait DecodeJResponse {
   def decodeJResponse(jRes: FullHttpResponse): Client.ClientResponse = {
     val status  = Status.fromHttpResponseStatus(jRes.status())
     val headers = Header.parse(jRes.headers())
-    Client.ClientResponse(status, headers, jRes.content())
+    val content =
+      Unpooled.buffer(jRes.content().readableBytes(), jRes.content().capacity()).writeBytes(jRes.content())
+    Client.ClientResponse(status, headers, content)
```

Comment:
As a general practice on the "backend" we don't use ZIO for writing any kind of server logic. The reason for doing that is primarily performance. ​What we can do in this case is move the logic to release content inside `decodeJResponse` instead.


The other things we might want to check is if the request doesn't have any content, don't waste CPU cycles in copying or releasing, instead use it as it is. Empty content AFAIK shouldn't cause leaks.

Reply:
- [ ] Check if release is happening automatically
- [ ] Optimize for perf by using the original content if it's empty.

Reply:
Release is not happening automatically.
@tusharmath please take a look again, I have updated the code for empty content case.